functions/ocr/app/main.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2018, Google, LLC.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START functions_ocr_setup]
+import base64
+import json
+import os
+
+from google.cloud import pubsub_v1
+from google.cloud import storage
+from google.cloud import translate
+from google.cloud import vision
+
+vision_client = vision.ImageAnnotatorClient()
+translate_client = translate.Client()
+publisher = pubsub_v1.PublisherClient()
+storage_client = storage.Client()
+
+project_id = os.environ['GCLOUD_PROJECT']
```

Comment:
I think we're discouraged from using this env variable; is this the variable present in production on GCF or is it for local use?

Reply:
I talked to Ace about the same question when I wrote the sample. Basically it is used because the Cloud Pub/Sub client library for Python explicitly requires the `project_id` parameter for the `topic_path` method (the Node.js client library, however, doesn't). Cloud Functions runtime does support it, though it is only documented in the [Firebase documentation](https://firebase.google.com/docs/functions/config-env).

Reply:
GCF support for env variables was documented here (added really recently) https://cloud.google.com/functions/docs/env-var

In short, env["GCP_PROJECT"] is preferred than env["GCLOUD_PROJECT"]

Reply:
@multiple1902 Thanks! I'll update it to use `GCP_PROJECT` instead.

Reply:
Sent in PR https://github.com/GoogleCloudPlatform/python-docs-samples/pull/1594